src/twisted/names/test/test_names.py

```diff
@@ -662,7 +682,56 @@ def test_recordMissing(self):
                 )
             ],
         )
-        self.assertEqual(additional, [])
+        self.assertEqual(resp.additional, [])
+
+    def test_queryNXDOMAInZone(self):
+        """
+        Requesting a record where no records exist for the name in question,
+        when the name belongs to the authoritative zone of the
+        L{FileAuthority}, results in a NXDOMAIN response with the SOA record
+        in the authority section.
+        """
+        testDomain = test_domain_com
+        testDomainName = b"nonexistent.subdomain." + testDomain.soa[0]
+
+        # The SOA record used when creating the testDomain
+        soa = soa_record
+
+        expected_resp = common.ResolverResponse(
```

Comment:
Minor twisted styleguide comment

```suggestion
        expectedResp = common.ResolverResponse(
```